### PR TITLE
FIX: Fixes #124 Bad data-url encoding causes CMS page-search bug.

### LINKED ISSUE #124: CMS Page Filter is broken
If I install Fluent in SS 3.1.12, I can no longer filter on PageTypes or perform a standard "Content" text search from within the CMS' "Filter" located in the centre pane. Instead, a complete list of _all_ page objects is returned as if I had instructed the CMS to build me the standard SiteTree. It isn't even a search-result that I can "clear".

I can try and futz the URL to include/exclude the "l" parameter to no avail e.g. 

`/?locale=en_NZ&q[FilterClass]=CMSSiteTreeFilter_Search&q[ClassName]=MyPageType`

or 

`/?locale=en_NZ&l=en_NZ&q[FilterClass]=CMSSiteTreeFilter_Search&q[ClassName]=MyPageType`

Removing the module, and it works again. By "Remove" I mean delete the "fluent" directory.

### PR DESCRIPTION
**Pull request by phptek**

### CODE REVIEW

**Comment by phptek** [comment]
Excuse formatting issues. My IDE is setup +-10% of SS PHP coding guidelines. Not sure what indentation you're using for JS :-)

**Comment by tractorcow** [comment]
tabs :D

**Comment by phptek** [comment]
Boooo

File: javascript/fluent.js [hunk 1]
```diff
@@ -97,17 +97,19 @@
...
-					newUrl = urlForLocale(url, $.cookie('FluentLocale_CMS'));
+        $('.cms-panel-deferred').entwine({
+            /**
+             * Ensure that any deferred panel URLs include the locale parameter in their URL
+             */
+            onadd: function () {
+                var url = this.attr('data-url'),
+                    newUrl = urlForLocale(url, $.cookie('FluentLocale_CMS')),
+                    // Ensure "incoming" data-url is properly filtered before being re-applied into the DOM
+                    i18nNewUrl = ss.i18n.sprintf(this.attr('data-url'), newUrl);
```
Comment (tractorcow) [inline javascript/fluent.js]: So we build the url from data-url and locale, and then sprintf the result back into data-url again?

**Comment by tractorcow** [comment]
Sorry, reading through https://github.com/tractorcow/silverstripe-fluent/issues/124 now :D

**Pull request closed without merging**